### PR TITLE
Add detect-secrets as the tool to detect git commit secrets

### PR DESCRIPTION
**Pull request by paulsimonandrews**
https://eaflood.atlassian.net/browse/PSD-613

### CODE REVIEW

File: docs/developer-laptop-setup/install-detect-secrets.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+## Configuration
+
+A `pre-commit` configuration file should exist in every FFC git repository that contains the necessary information to run `detect-secrets`. See the [guide for creating FFC git repositories](../source-code.md).
+
+Your system will need to be configured to set up the git hooks for both currently cloned, and future cloned, FFC repositories.
+
+### Currently cloned FFC git repositories
+
+Set up the `pre-commit` git hook to run `detect-secrets` by run the following command **in every FFC repository you have cloned on your system**:
```
Comment (st3v3nhunt) [inline docs/developer-laptop-setup/install-detect-secrets.md]: typo: `...by run the...` ➡️ `...by running the...`

File: docs/secrets-management.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Secrets Management
+
+## Client-side git secret detection
+
+All developers are required to [install detect-secrets](developer-laptop-setup/install-detect-secrets.md)
+when working on open-source FFC git repositories in github.
```
Comment (st3v3nhunt) [inline docs/secrets-management.md]: typo: `github` ➡️ `GitHub`

**Review by st3v3nhunt — CHANGES_REQUESTED** [review]
As discussed, the initial running of the baseline needs `--exclude-files package-lock.json` adding.
A couple of typos spotted.

File: docs/source-code.md [hunk 3]
```diff
@@ -25,6 +25,30 @@ Stale reviews should be automatically dismissed.
...
+  hooks:
+  - id: detect-secrets
+    args: ['--baseline', '.secrets.baseline']
+    exclude: ^package-lock.json|^.secrets.baseline
+```
+
+2. Create and add the baseline file:
+
+```
+detect-secrets scan > .secrets.baseline
```
Comment (st3v3nhunt) [inline docs/source-code.md]: `detect-secrets scan > .secrets.baseline` ➡️ `detect-secrets scan --exclude-files package-lock.json > .secrets.baseline`

File: docs/source-code.md [hunk 4]
```diff
@@ -25,6 +25,30 @@ Stale reviews should be automatically dismissed.
...
+
+1. Add the configuration file called `.pre-commit-config.yaml` to the repository root directory with the following contents (to exclude additional files/paths from secret detection update the `exclude:` regex):
+
+```
+- repo: git@github.com:Yelp/detect-secrets
+  rev: v0.13.1
+  hooks:
+  - id: detect-secrets
+    args: ['--baseline', '.secrets.baseline']
+    exclude: ^package-lock.json|^.secrets.baseline
```
Comment (st3v3nhunt) [inline docs/source-code.md]: I don't think `.secrets.baseline` needs to be included in the exclude list, at least I've not needed to and the file hasn't been listed as being scanned. I saw this after running the scan with the verbose option turned on to list all scanned files (`-v`).

File: docs/developer-laptop-setup/install-detect-secrets.md (hunk 1 above)
Comment (paulsimonandrews) [reply docs/developer-laptop-setup/install-detect-secrets.md]: Done

File: docs/source-code.md (hunk 3 above)
Comment (paulsimonandrews) [reply docs/source-code.md]: Done

File: docs/source-code.md (hunk 4 above)
Comment (paulsimonandrews) [reply docs/source-code.md]: Done (removed the baseline from an exclude)

File: docs/secrets-management.md (hunk 2 above)
Comment (paulsimonandrews) [reply docs/secrets-management.md]: Done

**Review by st3v3nhunt — APPROVED** [review]

**Comment by st3v3nhunt** [comment]
LGTM 👍

**Pull request merged**